lib/rack/proxy.rb

```diff
@@ -7,6 +7,17 @@ module Rack
   class Proxy
     VERSION = "0.7.2"
 
+    HOP_BY_HOP_HEADERS = %w[
+      connection
+      keep-alive
+      proxy-authenticate
+      proxy-authorization
+      te
+      trailer
+      transfer-encoding
+      upgrade
+    ].freeze
```

Comment:
Using a `Set` (or a `Hash` of `true` as `puma` does) should be faster.

Reply:
Done here